includes/model/class-openedx-woocommerce-plugin-enrollment.php

```diff
@@ -0,0 +1,785 @@
+            'oer_order_id'     => sanitize_text_field( $_POST['oer_order_id'] ),
+        );
+
+        $oer_action = sanitize_text_field( $_POST['oer_action'] );
+
+        $this->save_eor( $post, $oerarr, $oer_action );
+    }
+
+    /**
+     * Creates a new post in the database and runs it through the save.
+     *
+     * @param array  $oerarr An array containing the eor info.
+     * @param string $oer_action The API action to perform once the wp process is done.
+     *
+     * @return object $post The post object.
+     */
+    public function insert_new( $oerarr, $oer_action = '' ) {
+        $this->unregister_save_hook();
+
+        $new_oer = array(
+            'post_content' => 'Created automatically by woocommerce to fullfill an order.',
+            'post_type'    => 'openedx_enrollment',
+        );
+        $post_id = wp_insert_post( $new_oer );
+        $post    = get_post( $post_id );
+
+        $this->save_eor( $post, $oerarr, $oer_action );
+        return $post;
+    }
+
+    /**
+     * Save openedx request based on the incomming args.
+     *
+     * @param post   $post The post object.
+     * @param array  $oerarr An array containing the eor info.
+     * @param string $oer_action The API action to perform once the wp process is done.
+     */
+    public function save_eor( $post, $oerarr, $oer_action ) {
+
+        $post_id = $post->ID;
+
+        $oer_course_id    = $oerarr['oer_course_id'];
+        $oer_email        = $oerarr['oer_email'];
+        $oer_username     = $oerarr['oer_username'];
+        $oer_mode         = $oerarr['oer_mode'];
+        $oer_request_type = $oerarr['oer_request_type'];
+        $oer_order_id     = $oerarr['oer_order_id'];
+
+        // We need to have all 3 required params to continue.
+        $oer_user_reference = $oer_email || $oer_username;
+        if ( ! $oer_course_id || ! $oer_user_reference || ! $oer_mode ) {
+            return;
+        }
+
+        // Update the $post metadata.
+        update_post_meta( $post_id, 'course_id', $oer_course_id );
+        update_post_meta( $post_id, 'email', $oer_email );
+        update_post_meta( $post_id, 'username', $oer_username );
+        update_post_meta( $post_id, 'mode', $oer_mode );
+        update_post_meta( $post_id, 'order_id', $oer_order_id );
+
+        if ( $oer_request_type === 'enroll' ) {
+            update_post_meta( $post_id, 'is_active', true );
+        }
+        if ( $oer_request_type === 'unenroll' ) {
+            update_post_meta( $post_id, 'is_active', false );
+        }
+
+        // Only update the post status if it has no custom status yet.
+        if ( $post->post_status !== 'eor-success' && $post->post_status !== 'eor-pending' && $post->post_status !== 'eor-error' ) {
+            $this->update_post_status( 'eor-pending', $post_id );
+        }
+
+        // Handle the eox-core API actions.
+        if ( 'save_no_process' === $oer_action ) {
```

Comment:
defining the actions here is also very confusing. Since we will be defining the action later according to the API endpoints we create at the openedx plugin, I believe we can remove this block of `process_requests`.